File: thbigmatome/app/models/cost_player.rb
class CostPlayer < ApplicationRecord
  belongs_to :cost
  belongs_to :player

  validates :normal_cost, numericality: { only_integer: true, greater_than_or_equal_to: 1 }, allow_blank: true
  validates :relief_only_cost, numericality: { only_integer: true, greater_than_or_equal_to: 1 }, allow_blank: true
  validates :pitcher_only_cost, numericality: { only_integer: true, greater_than_or_equal_to: 1 }, allow_blank: true
  validates :fielder_only_cost, numericality: { only_integer: true, greater_than_or_equal_to: 1 }, allow_blank: true
  validates :two_way_cost, numericality: { only_integer: true, greater_than_or_equal_to: 1 }, allow_blank: true
end

File: thbigmatome-front/src/types/costPlayer.ts
export interface CostPlayer {
  id: number
  number: string | null
  name: string
  player_types: { id: number; name: string }[]
  normal_cost: number | null
  relief_only_cost: number | null
  pitcher_only_cost: number | null
  fielder_only_cost: number | null
  two_way_cost: number | null
  available_cost_types?: string[]
  player_card_id?: number | null
  variant?: string | null
  card_label?: string | null
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  [key: string]: any
}
